metals/src/main/scala/scala/meta/internal/metals/MultilineStringFormattingProvider.scala

```diff
@@ -126,28 +125,54 @@ final class MultilineStringFormattingProvider(
       textId: TextDocumentIdentifier,
       range: Range,
       newlineAdded: Boolean
-  )(
-      fn: (String, meta.Position) => List[TextEdit]
-  ): Future[List[TextEdit]] = Future {
-    val source = textId.getUri.toAbsolutePath
-    if (source.exists) {
-      val sourceText = buffer.get(source).getOrElse("")
-      val pos = range.getStart.toMeta(
-        Input.VirtualFile(source.toString(), sourceText)
-      )
-      if (pipeInScope(pos, sourceText, newlineAdded)) {
-        val tokens =
-          Input.VirtualFile(source.toString(), sourceText).tokenize.toOption
-        tokens.toList
-          .filter(multilineStringInTokens(_, pos, sourceText))
-          .flatMap(_ => fn(sourceText, pos))
+  )(fn: (String, meta.Position) => List[TextEdit]): Future[List[TextEdit]] =
+    Future {
+      val source = textId.getUri.toAbsolutePath
+      if (source.exists) {
+        val sourceText = buffer.get(source).getOrElse("")
+        val pos = range.getStart.toMeta(
+          Input.VirtualFile(source.toString(), sourceText)
+        )
+        if (pipeInScope(pos, sourceText, newlineAdded)) {
+          val tokens =
+            Input.VirtualFile(source.toString(), sourceText).tokenize.toOption
+          tokens.toList
+            .filter(multilineStringInTokens(_, pos, sourceText))
+            .flatMap(_ => fn(sourceText, pos))
+        } else Nil
       } else Nil
-    } else Nil
+    }
+
+  private def formatPipeLine(
+      line: Int,
+      lines: Array[String],
+      defaultIndent: String
+  ): TextEdit = {
+    val zeroPos = new Position(line, 0)
+    val lineText = lines(line)
+    val firstChar = lineText.trim.headOption
+
+    firstChar match {
+      case Some(char) if charIsPipe(char) =>
```

Comment:
nit: literal patterns could make this a bit nicer to read

```suggestion
      case Some('|') =>
```

Reply:
Ahh, that is simpler. Good call.